vllm/model_executor/model_loader/veturboio.py

```diff
@@ -0,0 +1,119 @@
+    num_thread: Optional[int] = 32
+    use_pinmem: Optional[bool] = False
+    use_direct_io: Optional[bool] = False
+    use_cipher: Optional[bool] = False  # not implemented yet
+
+    def _construct_veturboio_args(self) -> "VeturboIOArgs":
+        veturboio_args = {
+            "map_location": self.map_location,
+            "enable_fast_mode": self.enable_fast_mode,
+            "num_thread": self.num_thread,
+            "use_pinmem": self.use_pinmem,
+            "use_direct_io": self.use_direct_io,
+            "use_cipher": self.use_cipher,
+        }
+        return VeturboIOArgs(**veturboio_args)  # type: ignore
+
+    def verify_with_parallel_config(
+        self,
+        parallel_config: "ParallelConfig",
+    ) -> None:
+        if parallel_config.tensor_parallel_size > 1:
+            raise ValueError(
+                "veturboio does not support tensor parallelism yet")
+
+    def verify_with_model_config(self, model_config: "ModelConfig") -> None:
+        if model_config.quantization is not None:
+            logger.warning(
+                "Loading a model using VeturboIO with quantization on vLLM"
+                " is unstable and may lead to errors.")
+
+
+class VeturboIOAgent:
+
+    def __init__(self, veturboio_config: VeturboIOConfig):
+        if veturboio_error_msg is not None:
+            raise ImportError(
+                "VeturboIO is not installed. Please install ImportError "
+                "to use this feature with `pip install vllm[veturboio]`. "
+                "Error message: {}".format(veturboio_error_msg))
+
+        self.veturboio_config = veturboio_config
+        self.veturboio_args = (
+            self.veturboio_config._construct_veturboio_args())
+
+    def deserialize(self, model):
+        assert isinstance(model, torch.nn.Module)
+        assert self.veturboio_config.model_files is not None
+        start = time.perf_counter()
+        for model_file in self.veturboio_config.model_files:
+
+            tensors_dict = veturboio.load(
+                model_file,
+                helper=helper,
+                **self.veturboio_args.deserializer_params)
+            model.load_weights(iter(tensors_dict.items()))
+            del tensors_dict
+            # gc.collect()  # do gc collect immediately
+            torch.cuda.empty_cache()
+
+        end = time.perf_counter()
+        duration = end - start
+        logger.info("Deserialized model in %0.2fs by VeturboIO", duration)
+
+
+def load_with_veturboio_into_model(veturboio_config: VeturboIOConfig,
+                                   model: nn.Module):
+    assert veturboio_config.model_files is not None, (
+        "model files can not be None, "
+        "when load with veturboIO")
+    veturboio = VeturboIOAgent(veturboio_config)
+    return veturboio.deserialize(model)
+
+
+@dataclass
+class VeturboIOArgs:
+    map_location: Optional[str] = "cpu"
+    enable_fast_mode: Optional[bool] = True
+    num_thread: Optional[int] = 32
```

Comment:
what's the best configuration here? Should it detect the cpus available and use it for `num_thread`

Reply:
This default value `32` was copied from the veturbio implementation. Setting based on the number of available CPUs seems to be a better solution. I will make some changes to the implementation here later.

Reply:
I have tried `os` `psuitil` libraries, only `len(os.sched_getaffinity(0))` can detect the available cpu that provided by `--cpuset-cpus="1,2,3"`, but it does not work in the scenario of ` --cpu=4 `.

Is there a reliable way to obtain the CPU limit in the container?

![image](https://github.com/user-attachments/assets/ceb274b9-ea9d-4e03-a973-c0d5fad38d70)

Reply:
Or should we set 'num_thread' to None and contribute ` dynamically set num_trhead` to veturboio

Reply:
> Or should we set 'num_thread' to None and contribute  dynamically set num_trhead to veturboio
this sounds good to me.

Reply:
You can use `cat /sys/fs/cgroup/cpu/cpu.cfs_quota_us`. Let me also check any python util can provide accurate info.

Reply:
> > Or should we set 'num_thread' to None and contribute  dynamically set num_trhead to veturboio
> 
> this sounds good to me.这对我来说听起来不错。

pushed